llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -23299,6 +23299,110 @@ static SDValue LowerVSETCC(SDValue Op, const X86Subtarget &Subtarget,
     }
   }
 
+  // We get bad codegen for v8i32 compares on avx targets (without avx2) so if
+  // possible convert to a v8f32 compare.
+  if (VTOp0.getVectorElementType() == MVT::i32 && VTOp0 == MVT::v8i32 &&
```

Comment:
Just test VTOp0 == MVT::v8i32 for ? VTOp0.getVectorElementType() == MVT::i32 looks unnecessary.